habitat/tasks/rearrange/sub_tasks/nav_to_obj_task.py

```diff
@@ -0,0 +1,307 @@
+                self._config,
+                self._sim,
+            )
+
+        episode_id = sim.ep_info["episode_id"]
+
+        if self.force_obj_idx is not None:
+            full_key = f"{episode_id}_{self.force_obj_idx}"
+            if full_key in self.start_states:
+                (
+                    self._nav_target_pos,
+                    self._nav_target_angle,
+                ) = self.start_states[full_key]
+            else:
+                abs_true_point, task_cls_name, task_args = get_nav_from_obj_to(
+                    self.nav_obj_name, self.force_obj_idx, sim
+                )
+                targ_pos, self._nav_target_angle = self._get_nav_targ(
+                    task_cls_name, task_args, episode
+                )
+                orig_nav_targ_pos = sim.get_nav_pos(targ_pos)
+                self._nav_target_pos = np.array(
+                    sim.pathfinder.snap_point(orig_nav_targ_pos)
+                )
+                self._nav_target_angle = float(self._nav_target_angle)
+
+                self.start_states[full_key] = (
+                    self._nav_target_pos,
+                    self._nav_target_angle,
+                )
+                self.cache.save(self.start_states)
+            start_pos, start_rot = get_robo_start_pos(
+                sim, self._nav_target_pos
+            )
+        else:
+            if (
+                episode_id in self.start_states
+                and not self._config.FORCE_REGENERATE
+            ):
+                (
+                    self._nav_target_pos,
+                    self._nav_target_angle,
+                    start_pos,
+                    start_rot,
+                ) = self.start_states[episode_id]
+
+                sim.robot.base_pos = mn.Vector3(
+                    start_pos[0],
+                    sim.robot.base_pos[1],
+                    start_pos[2],
+                )
+                sim.robot.base_rot = start_rot
+            else:
+                (
+                    self._nav_target_pos,
+                    self._nav_target_angle,
+                    start_pos,
+                    start_rot,
+                ) = self._generate_nav_start_goal(episode)
+                self.start_states[episode_id] = (
+                    self._nav_target_pos,
+                    self._nav_target_angle,
+                    start_pos,
+                    start_rot,
+                )
+                self.cache.save(self.start_states)
+
+            targ_idxs, goal_pos = sim.get_targets()
+
+        observations = super().reset(episode)
+
+        if not sim.pathfinder.is_navigable(self._nav_target_pos):
+            print("Goal is not navigable")
+
+        if self._config.DEBUG_GOAL_POINT:
+            sim.viz_ids["nav_targ_pos"] = sim.visualize_position(
+                self._nav_target_pos,
+                sim.viz_ids["nav_targ_pos"],
+                r=10.0,
+            )
+
+        return observations
+
+
+def get_nav_targ(
```

Comment:
These methods look generic enough, maybe they should be moved to a more generic file like a navigation utility.